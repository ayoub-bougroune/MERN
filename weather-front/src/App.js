import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Container from '@material-ui/core/Container';
import Switch from '@material-ui/core/Switch';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { BrowserRouter } from 'react-router-dom';
import { Route, Switch as SwitchRouter} from "react-router-dom";

import Home from './Components/home'
import Detail from './Components/detail'
import AddCity from './Components/add-city';
import { getThemeProps } from '@material-ui/styles';
import API from './utils/API';

const drawerWidth = 240;
const useStyles = makeStyles(theme => {
  return (
  {
    root: {
        display: 'flex',
      },
      toolbar: {
        paddingRight: 24, // keep right padding when drawer closed
      },
      toolbarIcon: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'flex-end',
        padding: '0 8px',
        ...theme.mixins.toolbar,
      },
      appBar: {
        backgroundColor: theme.palette.secondary.dark,
        zIndex: theme.zIndex.drawer + 1,
        transition: theme.transitions.create(['width', 'margin'], {
          easing: theme.transitions.easing.sharp,
          duration: theme.transitions.duration.leavingScreen,
        }),
      },
      
      menuButton: {
        marginRight: 36,
      },
      menuButtonHidden: {
        display: 'none',
      },
      title: {
        flexGrow: 1,
      },
      headerLogo: {
        padding: '0 15px',
        backgroundColor: '#dedede',
        transform: 'translateX(24px)',
      },
      drawerPaper: {
        position: 'relative',
        whiteSpace: 'nowrap',
        width: drawerWidth,
        transition: theme.transitions.create('width', {
          easing: theme.transitions.easing.sharp,
          duration: theme.transitions.duration.enteringScreen,
        }),
      },
      drawerPaperClose: {
        overflowX: 'hidden',
        transition: theme.transitions.create('width', {
          easing: theme.transitions.easing.sharp,
          duration: theme.transitions.duration.leavingScreen,
        }),
        width: theme.spacing(7),
        [theme.breakpoints.up('sm')]: {
          width: theme.spacing(9),
        },
      },
      appBarSpacer: theme.mixins.toolbar,
      content: {
        flexGrow: 1,
        height: '100vh',
        overflow: 'auto',
        //backgroundColor: theme.palette.secondary.main
         background: `linear-gradient( ${theme.palette.secondary.light} , ${theme.palette.secondary.main} )`,
       // background: `linear-gradient(${theme.palette.secondary.ligth},${theme.palette.secondary.main})`,
      },
      container: {
        paddingTop: theme.spacing(4),
        paddingBottom: theme.spacing(4),
      },
      paper: {
        padding: theme.spacing(2),
        display: 'flex',
        overflow: 'auto',
        flexDirection: 'column',
      },
    });
  });
    



export default function App(props) {
  const classes = useStyles();
  const [checked, setChecked] = React.useState(props.check);
  const myAPI = new API();

  const handleChangeChecked = () =>{
    setChecked(!checked);
    props.handleChangeTheme(!checked? 'dark':'light');
  }
  const logOut = () => {
    myAPI.logout();
    window.location = "/login";
  }
 return (
    <React.Fragment>
      <div  className={classes.root}>
      <CssBaseline />
      <AppBar position="absolute" className={classes.appBar}>
        <Toolbar className={classes.toolbar}>        
         
          <Typography component="h1" variant="h6" color="inherit" noWrap className={classes.title}>
            Minimis
          </Typography>
          LIGHT 
          <FormControlLabel
            control={<Switch checked={checked} onChange={handleChangeChecked} />}
            label="DARK"
          />
         <Button href="#" color="primary" variant="outlined" onClick={logOut}>
            Log out
          </Button>
        </Toolbar>
      </AppBar>
      
      <main className={classes.content}>
      
        <div className={classes.appBarSpacer} />

          <Container maxWidth="lg" className={classes.container}>
            <Grid  container spacing={3}>
              <Grid item xs={12}>   
              <BrowserRouter>
                <SwitchRouter>
                  <Route exact path="/home" component={Home} />
                  <Route exact path="/detail/:id" component={Detail} />
                  <Route exact path="/add" component={AddCity} />
                </SwitchRouter> 
              </BrowserRouter>       
              </Grid>
            </Grid>  
          </Container>
        
      </main>
      
      </div>  
  
    </React.Fragment>
  );
}