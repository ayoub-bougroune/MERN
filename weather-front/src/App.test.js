import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const app = renderer
    .create(<App />);
  const tree = app.toJSON();
  expect(tree).toMatchSnapshot();
});
