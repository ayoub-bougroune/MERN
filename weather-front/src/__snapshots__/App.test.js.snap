// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionAbsolute makeStyles-appBar-4 MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <h1
        className="MuiTypography-root makeStyles-title-7 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        Minimis
      </h1>
      LIGHT
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          className="MuiSwitch-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-103 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                className="PrivateSwitchBase-input-106 MuiSwitch-input"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="MuiSwitch-thumb"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          DARK
        </span>
      </label>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Log out
        </span>
      </a>
    </div>
  </header>
  <main
    className="makeStyles-content-12"
  >
    <div
      className="makeStyles-appBarSpacer-11"
    />
    <div
      className="MuiContainer-root makeStyles-container-13 MuiContainer-maxWidthLg"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        />
      </div>
    </div>
  </main>
</div>
`;
